conda_smithy/templates/build_utils.py.tmpl

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+
+# This file was generated automatically from conda-smithy. To update this configuration,
+# update the conda-forge.yml and/or the recipe/meta.yaml.
+try:
+    from ruamel_yaml import safe_load, safe_dump
+except ImportError:
+    from yaml import safe_load, safe_dump
+import os
+import subprocess
+import json
+import argparse
+
+global_config = json.loads('''
+{{ channels | tojson }}''')
+call = subprocess.check_call
+
+
+def setup_conda_rc(args):
+    specific_config = safe_load(open(args.config_file))
+    if 'channel_sources' in specific_config:
+        # Due to rendering we may have more than one row for channel_sources, if nothing gets zipped with it
+        first_row = specific_config['channel_sources'][0]  # type: str
+        channels = [c.strip() for c in first_row.split(',')]
+    else:
+        channels = global_config['sources']
+
+    call(['conda', 'config', '--remove', 'channels', 'defaults'])
+    for c in reversed(channels):
+        call(['conda', 'config', '--add', 'channels', c])
+
+    call(['conda', 'config', '--set', 'show_channel_urls', 'true'])
+
+
+def upload_package(args):
+    specific_config = safe_load(open(args.config_file))
+    if 'channel_targets' in specific_config:
+        channels = [c.strip().split(' ') for c in specific_config['channel_targets']]
+    else:
+        channels = global_config['targets']
+
+    for owner, channel in channels:
+        call(['{{ upload_script }}', args.recipe_root, owner, '--channel=%s' % channel, '-m', args.config_file])
+
+
+def make_build_number(args):
+    specific_config = safe_load(open(args.config_file))
+    build_number_inc = int(specific_config.get("build_number_increment", [0])[0])
+    # nothing to do here
+    if build_number_inc == 0:
+        return
+
+    import re
+    recipe_dst = []
+    build_number_found = False
+    # Since recipes can contain build numbers in different outputs we have to do this gross part
+    with open(os.path.join(args.recipe_root, 'meta.yaml')) as recipe_src:
+        pat = re.compile("(\s*number\s*:\s*)(\d+)")
+        for line in recipe_src.readlines():
+            match = pat.match(line)
+            if match:
+                prefix, suffix = match.groups()
+                line = prefix + str(int(suffix) + build_number_inc) + '\n'
+                build_number_found = True
+            recipe_dst.append(line)
+
+    if build_number_found:
+        with open(os.path.join(args.recipe_root, 'meta.yaml'), 'w') as fo:
+            fo.writelines(recipe_dst)
+    else:
+        # conda build ignores clobber files that don't exist, so this is safe
+        config_dir, filename = os.path.split(args.config_file)
+        with open(os.path.join(config_dir, 'clobber_' + filename), 'w') as fo:
+            safe_dump({'build': {'number': 0 + build_number_inc}}, fo)
+
+
+def mangle_compiler(args):
+    """Try hard to break the compilers for osx"""
+    # TODO
```

Comment:
What are some good suggestions for things to do here to break compilers?

Reply:
Basically something we can do to make osx NOT have compilers that are found.  Essentially this is so that we don't accidentally build with system compilers when we want to build with the new ones.